lang/py/src/avro/schema.py

```diff
@@ -717,12 +819,27 @@ def make_avsc_object(json_data, names=None):
   """
   if names == None:
     names = Names()
-  
+
   # JSON object (non-union)
   if hasattr(json_data, 'get') and callable(json_data.get):
     type = json_data.get('type')
     other_props = get_other_props(json_data, SCHEMA_RESERVED_PROPS)
+    logical_type = None
+    if 'logicalType' in json_data:
```

Comment:
why would this be the case? (that 'logicalType' is in the json string)

Reply:
json_data is actually a python dictionary created by doing `json.loads` on the schema definition (I know, confusing naming convention). So you'd have something like `{"type": "long", "logicalType": "timestamp-micros"}`.